jiant/tasks/lm.py

```diff
@@ -172,3 +178,102 @@ def __init__(self, path, *args, **kw):
             "val": os.path.join(path, "valid.sentences.txt"),
             "test": os.path.join(path, "test.sentences.txt"),
         }
+
+
+@register_task("mlm", rel_path="WikiText103/")
+class MaskedLanguageModelingTask(Task):
+    """
+    Masked language modeling task on Wikipedia dataset
+    Attributes:
+        max_seq_len: (int) maximum sequence length
+        min_seq_len: (int) minimum sequence length
+        files_by_split: (dict) files for three data split (train, val, test)
+    """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        """Init class
+        Args:
+            path: (str) path that the data files are stored
+            max_seq_len: (int) maximum length of one sequence
+            name: (str) task name
+        """
+        super().__init__(name, **kw)
+        self.scorer1 = Average()
+        self.scorer2 = None
+        self._label_namespace = self.name + "_labels"
+        self.val_metric = "%s_perplexity" % self.name
+        self.val_metric_decreases = True
+        self.max_seq_len = max_seq_len
+        self.min_seq_len = 0
+        self.target_indexer = {"words": SingleIdTokenIndexer(namespace="tokens")}
+        self.files_by_split = {
```

Comment:
I'm a little confused by the versions of WikiText103 data that are floating around*. Here we're using a version with `train.txt`, `valid.txt` and `test.txt`; for the [`wiki103-classif` task](https://github.com/nyu-mll/jiant/blob/master/jiant/tasks/tasks.py#L2129-L2144) we depend on `train.sentences.txt`, `valid.sentences.txt`, and `test.sentences.txt`. Can you clarify what version of the data is expected here (in the filename/comments). Can you please also upload the final version of this data to the shared data repo?

\* Side note: there are two more versions of this data in the shared data repo with different naming patterns.

Reply:
Phu and I are using a raw, unpreprocessed Wikipedia dataset

Reply:
We'll make that clear.